src/pybind/mgr/orchestrator.py

```diff
@@ -694,26 +694,57 @@ class InventoryDevice(object):
     is permitted to support no extended properties (only normal block
     devices)
     """
-    def __init__(self):
-        self.blank = False
-        self.type = None  # 'ssd', 'hdd', 'nvme'
-        self.id = None  # unique within a node (or globally if you like).
-        self.size = None  # byte integer.
-        self.rotates = False # indicates if it is a spinning disk
-        self.available = False # can be used to create a new OSD?
-        self.dev_id = None  # vendor/model
-        self.extended = {}  # arbitrary JSON-serializable object
+    def __init__(self, blank=False, type=None, id=None, size=None,
+                 rotates=False, available=False, dev_id=None, extended=None,
+                 metadata_space_free=None):
+        # type: (bool, str, str, int, bool, bool. str, dict, bool) -> None
+
+        self.blank = blank
+
+        #: 'ssd', 'hdd', 'nvme'
+        self.type = type
+
+        #: unique within a node (or globally if you like).
+        self.id = id
+
+        #: byte integer.
+        self.size = size
+
+        #: indicates if it is a spinning disk
+        self.rotates = rotates
+
+        #: can be used to create a new OSD?
+        self.available = available
+
+        #: vendor/model
+        self.dev_id = dev_id
+
+        #: arbitrary JSON-serializable object
+        self.extended = extended if extended is not None else extended
 
         # If this drive is not empty, but is suitable for appending
         # additional journals, wals, or bluestore dbs, then report
         # how much space is available.
-        self.metadata_space_free = None
+        self.metadata_space_free = metadata_space_free
 
     def to_json(self):
         return dict(type=self.type, blank=self.blank, id=self.id,
                     size=self.size, rotates=self.rotates,
                     available=self.available, dev_id=self.dev_id,
-                    **self.extended)
+                    extended=self.extended)
+
+    @classmethod
+    def from_ceph_volume_inventory(cls, data):
+        dev = InventoryDevice()
+        dev.id = data["path"]
+        dev.type = 'hdd' if data["sys_api"]["rotational"] == "1" else 'sdd/nvme'
+        dev.size = data["sys_api"]["size"]
```

Comment:
why not
dev.size = ["sys_api"]["human_readable_size"] 
?

Reply:
We have to stick to the Interface. If rook, ssh and deepsea are adding an int in bytes, ansible cannot use that field as `human_readable_size`

Reply:
InventoryDevice predates c-v inventory. as described in InventoryDevice is supposed to be of type `int`. rook and deepsea already populate `size` as such.

Reply:
ok. Probably a human friendly output will be requested at some time... it can be produced easily.
LGTM

Reply:
Ive just added format_bytes: 

https://github.com/ceph/ceph/pull/26094/commits/81dc75be10a6cf648932b876b99f843284942f26#diff-dfbc76d00f82320b0e6ac732319aa545R768

Reply:
Thx!